x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/settings/ReloadRemoteClusterCredentialsAction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.settings;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.settings.Settings;
+
+import java.io.IOException;
+
+public class ReloadRemoteClusterCredentialsAction extends ActionType<ActionResponse.Empty> {
```

Comment:
nit: there's no need to subclass `ActionType` here. It's a common pattern (again, it made more sense in the distant past) but we're trying to remove it (see e.g.  #103055, #102800, #102354, #101053, #101037).

Reply:
Thanks for this pointer too! I've implemented your suggestion. There is a bit of dependency spaghetti that comes with it since the action type needs to be available in `xpack.core.security` whereas the transport action (which is where we'd normally store the type with this approach) is not available here. 

In particular, [SystemPrivilege](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/SystemPrivilege.java) requires the action name to register access to the action for the system user. I could duplicate the name literal there but that's error-prone and I prefer to avoid it. 

Security has so far followed the pattern of keeping transport actions under `xpack.security` and their corresponding action declarations in `xpack.core.security`, which I'd like to keep with. 

Out of the available options, a new `ActionTypes` class to store action types that fall under this category feels like the best approach.

Reply:
++ makes sense to me